mongomock/collection.py

```diff
@@ -254,6 +255,20 @@ def _copy_field(obj, container):
     return copy.copy(obj)
 
 
+def _validate_data_fields(data):
+    def recursive_none_key_check(data):
```

Comment:
This function isn't using any closure, so I'd rather not have it as a nested function